Sample pull request to identify the changes (#1)
ferralej/nf-tower-docs: docs/compute-envs/aws-batch.md

---
## Overview
!!! note "Disclaimer"
    This guide assumes you have an existing [Amazon Web Service (AWS) Account](https://aws.amazon.com/). Sign up for a free AWS account [here](https://portal.aws.amazon.com/billing/signup).

There are two ways to create a **Compute Environment** for **AWS Batch** with Tower.

> 1. **Tower Forge** for AWS Batch: This option automatically creates AWS Batch resources in your AWS account.

Author: ```suggestion
1. **Tower Forge** for AWS Batch: This option automatically manages the AWS Batch resources in your AWS account.
```

2. **Tower Launch**: It allows you to create a compute environment using existing AWS Batch resources.

If you don't have an AWS Batch environment fully set-up yet, it is suggested to follow the [Tower Forge](#forge) guide. 

If your administrator has provided you with an AWS queue, or if you have set up AWS Batch previously, please follow the [Tower Launch](#manual) guide.

## Forge

!!! warning
    Follow these instructions if you have **not** pre-configured an AWS Batch environment. Note that this will create resources in your AWS account that you may be charged for by AWS.

Tower Forge automates the configuration of an [AWS Batch](https://aws.amazon.com/batch/) compute environment and queues required for the deployment of Nextflow pipelines.

## Forge AWS Resources

### IAM User Permissions

> To use the Tower Forge feature, Tower requires an Identity and Access Management (IAM) user with the permissions listed in the following [policy file](https://github.com/seqeralabs/nf-tower-aws/blob/master/forge/forge-policy.json). These authorizations are more permissive than those required to only launch a pipeline since Tower needs to create AWS resources on your behalf.

Author: ```suggestion
To use the Tower Forge feature, Tower requires an Identity and Access Management (IAM) user with the permissions listed in the following [policy file](https://github.com/seqeralabs/nf-tower-aws/blob/master/forge/forge-policy.json). These authorizations are more permissive than those required to only launch a pipeline, since Tower needs to manage AWS resources on your behalf.
```

### Creating an S3 Bucket for Storage

S3 stands for Simple Storage Service and is a type of **object storage**. To access files and store the results for our pipelines, we have to create an **S3 Bucket** next.

We must grant our new Tower IAM user access to this bucket.

**1.** Navigate to [S3 service](https://console.aws.amazon.com/s3/home) 

**2.** Select **Create New Bucket**.

**3.** Enter a unique name for your Bucket and select a region.

![](_images/aws_create_bucket.png)

!!! warning "Which AWS region should I use?"
    The region of the bucket should be in the *same region as the compute environment which we will set in the next section*. Typically users select a region closest to their physical location but Tower Forge supports creating resources in any of the available AWS regions.

**4.** Select the default options for **Configure options**.

![](_images/aws_new_bucket_configure_options.png)

**5.** Select the default options for **Set permissions**.

![](_images/aws_new_bucket_set_permissions.png)

**6.** Review the bucket and select **Create bucket**.

![](_images/aws_new_bucket_review.png)

!!! warning  "S3 Storage Costs" 

>     S3 is used by Nextflow for the storage of intermediary files. For production pipelines, this can amount to a large quantity of data. To reduce costs, when configuring a bucket, users should consider management options such as the ability to automatically delete these files after 30 days. For more information on this process, click [here] (https://aws.amazon.com/de/premiumsupport/knowledge-center/s3-empty-bucket-lifecycle-rule/).

Author: The link was for a German locale 😉 

```suggestion
    S3 is used by Nextflow for the storage of intermediary files. For production pipelines, this can amount to a large quantity of data. To reduce costs, when configuring a bucket, users should consider management options such as the ability to automatically delete these files after 30 days. For more information on this process, click [here] (https://aws.amazon.com/premiumsupport/knowledge-center/s3-empty-bucket-lifecycle-rule/).
```
